### PR TITLE
got sorting filter working (front end and back end)

### PR DESCRIPTION
**Pull request by muse-gabs**

### CODE REVIEW

File: static/js/dashboard.js [hunk 1]
```diff
@@ -174,6 +174,14 @@ function toggleEditAdView(id){
     thisView.hidden = !thisView.hidden;
 }
 
+// this was me playing around with using a function to XMLHTTPrequest the
```
Comment (rorrell) [inline static/js/dashboard.js:176]: I'm guessing this should be deleted?

File: views/dashboard.handlebars [hunk 2]
```diff
@@ -25,6 +25,42 @@
...
+     <!--       <button class="button active" onclick="filterSelection('a.Title ASC')"> Ascending Title </button>
+                <button class="button" onclick="filterSelection('a.Title DESC')"> Descending Title </button>
+                <button class="button" onclick="filterSelection('a.DatePosted ASC')"> Most Recent </button>
+                <button class="button" onclick="filterSelection('a.DatePosted DESC')"> Least Recent </button>
+                <button class="button" onclick="filterSelection('a.LocationRadius ASC')"> Ascending Location Radius </button>
+                <button class="button" onclick="filterSelection('a.LocationRadius DESC')"> Descending Location Radius </button>
+                <button class="button" onclick="filterSelection('a.DateCreated ASC')"> Ascending Date Created </button>
+                <button class="button" onclick="filterSelection('a.DateCreated DESC')"> Descending Date Created </button> -->
+
+<!-- the stuff above was just me playing around with function and buttons, can be disregarded or deleted -->
```
Comment (rorrell) [inline views/dashboard.handlebars:47]: Does this line need to be deleted?

File: views/dashboard.handlebars [hunk 3]
```diff
@@ -25,6 +25,42 @@
...
+<!-- below is the actual form for sorting ads based on the filter, currently there's nothing that shows which filter they are set to, and I might have flipped
```
Comment (rorrell) [inline views/dashboard.handlebars:48]: I believe we do need to show what filter is currently set, both when they first come to the page (DatePosted DESC) and every time the sort order is changed.

File: views/dashboard.handlebars [hunk 4]
```diff
@@ -25,6 +25,42 @@
...
+
+<!-- the stuff above was just me playing around with function and buttons, can be disregarded or deleted -->
+<!-- below is the actual form for sorting ads based on the filter, currently there's nothing that shows which filter they are set to, and I might have flipped 
+     some of the ASC and DESC, but those are easy fixes. Things were breaking and fixing themselves and I was working on it from like 3am to 5am and then I lost internet 
+     so not really sure if I fixed the changes or not, plz double-check, k tanks -->
+                <form id="sort" action="/adSortOrder" method="post">
+                   <select name="sortOrder">
+                        <option value="a.Title ASC" selected> Ascending Title</option>
+                        <option value="a.Title DESC"> Descending Title </option>
+                        <option value="a.DatePosted ASC"> Most Recent</option>
```
Comment (rorrell) [inline views/dashboard.handlebars]: I believe the one above this comment and the one below are flipped.  Also you might want to call it "Most Recently Posted" and "Least Recently Posted".

File: views/dashboard.handlebars [hunk 5]
```diff
@@ -25,6 +25,42 @@
...
+                        <option value="a.DatePosted DESC">Least Recent </option>
+                        <option value="a.LocationRadius ASC"> Ascending Location Radius </option>
+                        <option value="a.LocationRadius DESC"> Descending Location Radius </option>
```
Comment (rorrell) [inline views/dashboard.handlebars]: I would recommend calling the two above "Closest" and "Farthest".  The meaning would then be clearer.

File: views/dashboard.handlebars [hunk 6]
```diff
@@ -25,6 +25,42 @@
...
+                        <option value="a.DateCreated ASC"> Ascending Date Created</option>
+                        <option value="a.DateCreated DESC"> Descending Date Created </option>
```
Comment (rorrell) [inline views/dashboard.handlebars]: I think the two above would make more sense as "Least Recently Created" (that's the ASC one) and "Most Recently Created" (that's the DESC one).

**Review by rorrell — CHANGES_REQUESTED** [review]
See my comments on the "Files Changed" tab.

File: views/dashboard.handlebars (hunk 4 above)
Comment (muse-gabs) [reply views/dashboard.handlebars]: fixed

File: views/dashboard.handlebars (hunk 5 above)
Comment (muse-gabs) [reply views/dashboard.handlebars]: fixed

File: views/dashboard.handlebars (hunk 6 above)
Comment (muse-gabs) [reply views/dashboard.handlebars]: fixed

File: views/dashboard.handlebars (hunk 3 above)
Comment (muse-gabs) [reply views/dashboard.handlebars:48]: fixed

**Pull request merged**